File: samples/Android/xtruntime/src/main/java/com/opensource/xtruntime/XTRDevice.kt
package com.opensource.xtruntime

import android.os.Build

/**
 * Created by cuiminghui on 2017/9/28.
 */
enum class DeviceOrientation {
    Unknown,
    Portrait,
    PortraitUpsideDown,
    LandscapeLeft,
    LandscapeRight,
    FaceUp,
    FaceDown,
}

class XTRDevice: XTRComponent() {

    override val name: String = "XTRDevice"

    companion object {
        internal var current: InnerObject? = null
    }

    fun xtr_current(): InnerObject {
        if (current == null) {
            current = InnerObject(xtrContext)
        }
        return current ?: InnerObject(xtrContext)
    }

    class InnerObject(val xtrContext: XTRContext) {

        fun xtr_name(): String {
            return Build.BRAND
        }

        fun xtr_systemName(): String {
            return "Android"
        }

        fun xtr_systemVersion(): String {
            return Build.VERSION.SDK_INT.toString()
        }

        fun xtr_xtRuntimeVersion(): String {
            return XTRuntime.version
        }

        fun xtr_model(): String {
            return Build.MODEL
        }

        internal var orientation = DeviceOrientation.Unknown

        fun xtr_orientation(): Int {
            return orientation.ordinal
        }

    }

}

File: ios/runtime/XTRDevice.m
//
//  XTRDevice.m
//  XTSample
//
//  Created by 崔明辉 on 2017/9/28.
//  Copyright © 2017年 UED Center, YY Inc. All rights reserved.
//

#import "XTRDevice.h"
#import "XTCore.h"
#import "XTContext.h"

@implementation XTRDevice

+ (NSString *)name {
    return @"XTRDevice";
}

+ (NSString *)xtr_name {
    return [UIDevice currentDevice].name;
}

+ (NSString *)xtr_systemName {
    return [UIDevice currentDevice].systemName;
}

+ (NSString *)xtr_systemVersion {
    return [UIDevice currentDevice].systemVersion;
}

+ (NSString *)xtr_xtRuntimeVersion {
    return [XTCore version];
}

+ (NSString *)xtr_model {
    return [UIDevice currentDevice].model;
}

+ (NSInteger)xtr_orientation {
    return [UIDevice currentDevice].orientation;
}

@end
